src/EventStore.Core/Services/Transport/Http/Controllers/AdminController.cs

```diff
@@ -41,16 +43,31 @@ private void OnPostShutdown(HttpEntityManager entity, UriTemplateMatch match)
 
         private void OnPostScavenge(HttpEntityManager entity, UriTemplateMatch match)
         {
-            if (entity.User != null && (entity.User.IsInRole(SystemRoles.Admins) || entity.User.IsInRole(SystemRoles.Operations)))
-            {
-                Log.Info("Request scavenging because /admin/scavenge request has been received.");
-                Publish(new ClientMessage.ScavengeDatabase(new NoopEnvelope(), Guid.Empty, entity.User));
-                entity.ReplyStatus(HttpStatusCode.OK, "OK", LogReplyError);
-            }
-            else
-            {
-                entity.ReplyStatus(HttpStatusCode.Unauthorized, "Unauthorized", LogReplyError);
-            }
+            Log.Info("Request scavenging because /admin/scavenge request has been received.");
+
+            var envelope = new SendToHttpEnvelope(_networkSendQueue, entity, (e, message) =>
+                {
+                    var completed = message as ClientMessage.ScavengeDatabaseResponse;
+                    return e.ResponseCodec.To(completed?.ScavengeId);
```

Comment:
@lscpike hi, would you be able to return `completed` instead of `completed?.ScavengeId` here so that it's serialized as JSON/XML depending on the `Content-Type` specified. It's just a just a matter of consistency among all the controllers. thanks again for this great PR :)

Reply:
I'm only returning the scavenge id and don't want to expose the whole ScavengeDatabaseResponse. As XML this comes back like this:

<?xml version="1.0" encoding="utf-8"?><string>2b10b34a-1cfe-4d5c-b48f-41eb985983ed</string>

If that's not good enough I'll add a DTO to wrap it. Thoughts?

Reply:
@lscpike thanks. yes, a DTO would be perfect. the only issue is with JSON - it just prints out the id with quotes which is not a valid json string.

Reply:
Done.

Reply:
@lscpike thanks!